File: frontend/types/interview.ts
export interface Interview {
  id: number
  candidateId: number
  candidateName?: string
  interviewerId?: number
  questionId: number
  questionTitle?: string
  language: string
  status: 'scheduled' | 'in_progress' | 'completed' | 'cancelled'
  scheduledAt?: string
  startedAt?: string
  completedAt?: string
  interviewLinkToken: string
  createdAt?: string
  updatedAt?: string
}

export interface CreateInterviewRequest {
  questionId: number
  candidateId: number
  language: string
  scheduledAt?: string
  interviewerId?: number
}

export interface CreateInterviewResponse {
  id: number
  candidateId: number
  questionId: number
  language: string
  status: string
  interviewLinkToken: string
  createdAt: string
}

export interface InterviewListResponse {
  id: number
  candidateName: string
  questionTitle: string
  status: string
  scheduledAt?: string
  startedAt?: string
  interviewLinkToken: string
}

export interface Question {
  id: number
  title: string
  description: string
  difficulty: string
  timeLimitMinutes: number
  supportedLanguages: string[]
  createdAt?: string
}

export interface Candidate {
  id: number
  name: string
  email: string
  createdAt?: string
}

File: backend/src/main/java/com/example/interviewAI/entity/Interview.java
package com.example.interviewAI.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDateTime;

@Entity
@Table(name = "interviews")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Interview {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = true)
    private String language; // 'java', 'python', 'javascript', etc. - Set when candidate submits setup

    @Column
    private LocalDateTime scheduledAt;

    @Column
    private LocalDateTime startedAt;

    @Column
    private LocalDateTime completedAt;

    @Column(nullable = false)
    private String status; // 'scheduled', 'in_progress', 'completed', 'cancelled'

    @Column(unique = true)
    private String interviewLinkToken;

    @Column(nullable = false)
    private LocalDateTime createdAt = LocalDateTime.now();

    @ManyToOne
    @JoinColumn(name = "company_id")
    private Company company;

    @ManyToOne
    @JoinColumn(name = "question_id")
    private Question question;

    @ManyToOne
    @JoinColumn(name = "candidate_id")
    private Candidate candidate;

    @ManyToOne
    @JoinColumn(name = "interviewer_id")
    private User interviewer;
}
